core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -428,12 +428,22 @@ int getSpeed( OSMWay way )
         // Until now we assumed that the way is no pushing section
         // Now we check, but only in case that our speed is bigger compared to the PUSHING_SECTION_SPEED
         if ((speed > PUSHING_SECTION_SPEED)
-                && (!way.hasTag("bicycle", intendedValues) && way.hasTag("highway", pushingSections)))
+                && (!way.hasTag("bicycle", intendedValues)))
         {
-            if (way.hasTag("highway", "steps"))
-                speed = PUSHING_SECTION_SPEED / 2;
+            if (way.hasTag("highway", pushingSections))
+            {
+                if (way.hasTag("highway", "steps"))
+                    speed = PUSHING_SECTION_SPEED / 2;
+                else
+                    speed = PUSHING_SECTION_SPEED;
+            }
             else
-                speed = PUSHING_SECTION_SPEED;
+            {
+                if (way.hasTag("bicycle", "dismount"))
```

Comment:
instead of the separate 'if' maybe just 
```java
if (way.hasTag("highway", pushingSections) || way.hasTag("bicycle", "dismount"))
```
this way we would handle the steps too?